crates/wasm-pkg-client/src/warg/config.rs

```diff
@@ -1,29 +1,46 @@
-use std::path::PathBuf;
+use std::{fmt::Debug, path::PathBuf, sync::Arc};
 
 use secrecy::{ExposeSecret, SecretString};
 use serde::{Deserialize, Serialize, Serializer};
+use warg_crypto::signing::PrivateKey;
 use wasm_pkg_common::{config::RegistryConfig, Error};
 
 /// Registry configuration for Warg backends.
 ///
 /// See: [`RegistryConfig::backend_config`]
-#[derive(Clone, Debug, Default, Serialize)]
+#[derive(Clone, Default, Serialize)]
 #[serde(into = "WargRegistryConfigToml")]
 pub struct WargRegistryConfig {
     /// The configuration for the Warg client.
     pub client_config: warg_client::Config,
     /// The authentication token for the Warg registry.
     pub auth_token: Option<SecretString>,
+    /// A signing key to use for publishing packages.
+    // NOTE(thomastaylor312): This couldn't be wrapped in a secret because the outer type doesn't
+    // implement Zeroize. However, the inner type is zeroized.
+    pub signing_key: Option<Arc<PrivateKey>>,
     /// The path to the Warg config file, if specified.
     pub config_file: Option<PathBuf>,
 }
 
+impl Debug for WargRegistryConfig {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("WargRegistryConfig")
+            .field("client_config", &self.client_config)
+            .field("auth_token", &self.auth_token)
```

Comment:
Should `"[redacted]"` for the `auth_token` as well?

Reply:
I'm letting the `Secret` type handle that for me. I would have still derived `Debug` because `PrivateKey` is just a `Secret` under the hood, but it didn't implement that for me. I can add that if you'd like, but this was mostly a workaround because `PrivateKey` isn't `Debug`